segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/ContainerMetadataUpdateTransaction.java

```diff
@@ -114,13 +114,8 @@
         this.newTruncationPoints = new ArrayList<>();
         this.newSegments = new HashMap<>();
         this.newSegmentNames = new HashMap<>();
-        if (baseMetadata.isRecoveryMode()) {
-            this.newSequenceNumber = ContainerMetadata.INITIAL_OPERATION_SEQUENCE_NUMBER;
-        } else {
-            this.newSequenceNumber = Long.MIN_VALUE;
-        }
-
         this.sealed = false;
+        resetNewSequenceNumber();
```

Comment:
One clarification here. The description of the pull request says:

> however due to a bug in the code, this was being generated off the
> newly assigned sequence number (which would only be set during
> recovery, and always starts off at Long.MIN_VALUE).

But if the base metadata is in recovery mode, then we set it to a different constant:

```
if (baseMetadata.isRecoveryMode()) {
             this.newSequenceNumber = ContainerMetadata.INITIAL_OPERATION_SEQUENCE_NUMBER;
 } else {
             this.newSequenceNumber = Long.MIN_VALUE;
 }
```